sympy/physics/continuum_mechanics/truss.py

```diff
@@ -0,0 +1,494 @@
+            The label of the node/member for which the label has
+            to be changed.
+
+        new_label: String or Symbol
+            The new label of the node/member.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.continuum_mechanics.truss import Truss
+        >>> t = Truss()
+        >>> t.add_node('A', 0, 0)
+        >>> t.add_node('B', 3, 0)
+        >>> t.nodes
+        [('A', 0, 0), ('B', 3, 0)]
+        >>> t.change_label('A', 'C')
+        >>> t.nodes
+        [('C', 0, 0), ('B', 3, 0)]
+        >>> t.add_member('BC', 'B', 'C')
+        >>> t.members
+        [['BC', 'B', 'C']]
+        >>> t.change_label('BC', 'BC_new')
+        >>> t.members
+        [['BC_new', 'B', 'C']]
+        """
+        if label not in self._node_labels and label not in self._member_labels:
+            raise ValueError("No such node/member exists")
+        if label in self._node_labels:
+            for node in self._nodes:
+                if node[0] == label:
+                    self._node_labels[self._node_labels.index(node[0])] = new_label
+                    self._loads[new_label] = self._loads[node[0]]
+                    self._loads.pop(node[0])
+                    self._supports[new_label] = self._supports[node[0]]
+                    self._supports.pop(node[0])
+                    if self._supports[new_label] == 'pinned':
+                        self._reaction_forces['R_'+str(new_label)+'_x'] = self._reaction_forces['R_'+str(label)+'_x']
+                        self._reaction_forces['R_'+str(new_label)+'_y'] = self._reaction_forces['R_'+str(label)+'_y']
+                    elif self._supports[new_label] == 'roller':
+                        self._reaction_forces['R_'+str(new_label)+'_y'] = self._reaction_forces['R_'+str(label)+'_y']
+                    for member in self._members:
+                        if member[1] == node[0]:
+                            member[1] = new_label
+                            self._member_nodes[member[0]] = [new_label, member[2]]
+                            self._nodes_occupied[(new_label, member[2])] = True
+                            self._nodes_occupied[(member[2], new_label)] = True
+                            self._nodes_occupied.pop(tuple([label, member[2]]))
+                            self._nodes_occupied.pop(tuple([member[2], label]))
+                        elif member[2] == node[0]:
+                            member[2] = new_label
+                            self._member_nodes[member[0]] = [member[1], new_label]
+                            self._nodes_occupied[(member[1], new_label)] = True
+                            self._nodes_occupied[(new_label, member[1])] = True
+                            self._nodes_occupied.pop(tuple([member[1], label]))
+                            self._nodes_occupied.pop(tuple([label, member[1]]))
+                    self._nodes[self._nodes.index((label, node[1], node[2]))] = (new_label, node[1], node[2])
+
```

Comment:
It would be better to have different change label functions for members and nodes. This way if some node and member share the same name, both labels won't get changed